Automattic/jetpack
# Jetpack: re-attempt to get the VideoPress video preview

This PR tries to handle the issue when the video preview is not returned by the `getEmbedPreview()` selector.
This issue happens (In my short experience) when the video is recently uploaded.

Taking a look at the network tab, the following screenshot shows how the first request responds with `404` and the second one with `200`. returning the preview to be used to render the video player.

<img width="1057" alt="image" src="https://user-images.githubusercontent.com/77539/176259467-5642f6cd-5c78-4c75-8f81-d38990a0a837.png">

I've updated the UI showing the attempt number when the preview is not defined initially. Take a look at the video of testing instructions.

Let's implement a better UI in follow-up PRs.

Fixes #

#### Changes proposed in this Pull Request:

* Jetpack: re-attempt to get the VideoPress video preview

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Go to block editor
* Add a new VideoPress block
* Upload a video
* Confirm the video player is always rendered. Sometimes, it happens in on subsequent attempts

https://user-images.githubusercontent.com/77539/176260047-8e76c9c7-79a4-45af-a8d8-af1fe1139212.mov

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202514379845434
